tests/Unit/Utilities/Test_TaggedTuple.cpp

```diff
@@ -86,6 +86,16 @@ SPECTRE_TEST_CASE("Unit.Utilities.TaggedTuple", "[Utilities][Unit]") {
   }
 
   test_serialization(test2);
+
+  /// [reorder_tagged_tuple_example]
+  const auto test3 = tuples::reorder_tagged_tuple<
```

Comment:
You should add a test where you pass a non-const lvalue to the function (it should hold something that breaks during move, like a std::vector) and verify that the input is still valid after the call. That'll expose the bug you currently have.